cmd/docker.sh

```diff
@@ -17,15 +17,27 @@ register 'compose' 'ps' 'list containers' compose_ps
 function compose_top(){ compose_exec top $@; }
 register 'compose' 'top' 'display the running processes of a container' compose_top
 
-# the 'docker compose' subcommand is now the recommended method of calling compose.
-# if not available, we fallback to the legacy 'docker-compose' command.
 function compose_exec(){
+  # check whether there is a local compose file and append them to the global one.
+  composeFilePath="-f ../../docker-compose.yml -f ./docker-compose.yml"
+  export peliasRegion="projects/$(basename $(pwd))"
+  if [ ! -f docker-compose.yml ];then
+    echo "No local file. Using only global compose file."
+    composeFilePath="-f ../../docker-compose.yml"
+  fi
+
+  # the 'docker compose' subcommand is now the recommended method of calling compose.
+  # if not available, we fallback to the legacy 'docker-compose' command.
   NATIVE_COMPOSE_VERSION=$(docker compose version 2> /dev/null || true)
   if [ -z "$NATIVE_COMPOSE_VERSION" ]; then
-    docker-compose $@;
+    dockerComposeCommand="docker-compose"
   else
-    docker compose $@;
+  dockerComposeCommand="docker compose"
   fi
+  
+  # execute the command with the compose file(s) and any additional arguments
+  $dockerComposeCommand $composeFilePath $@;
```

Comment:
I'm not that happy with the solution, since it's not very readable. However, it's the best way I could think of to avoid some code duplication.